classes/OpenXdmod/Setup/AddResourceSetup.php

```diff
@@ -77,23 +98,64 @@ public function handle()
 
-        $nodes = $this->console->prompt('How many nodes does this resource have?');
+        $nodes = $this->console->prompt('How many CPU nodes does this resource have?');
         if (empty($nodes) || !is_numeric($nodes)) { $nodes = 1; }
 
-        $cpus = $this->console->prompt('How many total processors (cpu cores) does this resource have?');
+        $cpus = $this->console->prompt('How many total CPU processors (cpu cores) does this resource have?');
         if (empty($cpus) || !is_numeric($cpus)) { $cpus = 1; }
 
-        $ppn = $cpus / $nodes;
+        $ppn = ($nodes == 0) ? 0 : $cpus / $nodes;
+
+        if ($resource_allocation_type == 'gpu' || $resource_allocation_type == 'gpu_node') {
+            $gpu_nodes = $this->console->prompt('How many GPU nodes does this resource have?');
+            if (empty($gpu_nodes) || !is_numeric($gpu_nodes)) {
+                $gpu_nodes = 0;
+            }
+
+            $gpus = $this->console->prompt('How many total GPU processors (gpu cores) does this resource have?');
+            if (empty($gpus) || !is_numeric($gpus)) {
+                $gpus = 0;
+            }
+
+            $gpu_ppn = ($gpu_nodes == 0) ? 0 : $gpus / $gpu_nodes;
+        }
 
         $this->parent->addResource(
             array(
                 'resource'   => $resource,
                 'name'       => $name,
                 'type'       => $type,
-                'processors' => (int)$cpus,
-                'nodes'      => (int)$nodes,
-                'ppn'        => (int)$ppn,
+                'resource_allocation_type'    => $resource_allocation_type,
+                'cpu_processor_count' => (int)$cpus,
+                'cpu_node_count'      => (int)$nodes,
+                'cpu_ppn'             => (int)$ppn,
+                'gpu_processor_count' => (int)$gpus,
+                'gpu_node_count'      => (int)$gpu_nodes,
+                'gpu_ppn'             => (int)$gpu_ppn,
+                'start_date'          => $resource_start_date
             )
         );
     }
+
+    /**
+    * Prompt the user for the start date of a resource. If left empty it will default to the current day.
+    * It also validates that any date string entered is a valid date and in the correct format.
+    *
+    * @return string $resource_start_date string
+    */
+    public function getResourceStartDate()
+    {
+        $resource_start_date  = $this->console->prompt('Resource Start Date, in YYYY-mm-dd format', date('Y-m-d'));
+        $resource_start_date_parsed = DateTime::createFromFormat("Y-m-d", $resource_start_date);
+
+        if (empty($resource_start_date)) {
+            $resource_start_date_parsed = DateTime::createFromFormat("Y-m-d", date('Y-m-d'));
+        }
+        elseif ($resource_start_date_parsed === false) {
+            $this->console->displayMessage("The date you entered is in the wrong format. Please enter the date in YYYY-mm-dd format.");
+            $this->getResourceStartDate();
```

Comment:
```suggestion
            return $this->getResourceStartDate();
```
Otherwise it falls through and tries to call `->format` on `false`.